python/lsst/daf/butler/registry/bridge/ephemeral.py

```diff
@@ -54,6 +54,9 @@ def insert(self, refs: Iterable[DatasetIdRef]) -> None:
         # Docstring inherited from DatastoreRegistryBridge
         self._datasetIds.update(ref.getCheckedId() for ref in refs)
 
+    def forget(self, refs: Iterable[DatasetIdRef]) -> None:
+        self._datasetIds.difference_update({ref.id for ref in refs})
```

Comment:
I think there is no need to make a set first, `difference_update` accepts any iterable.